internal/handlers/internal/secrets/secrets_test.go

```diff
@@ -0,0 +1,75 @@
+package secrets_test
+
+import (
+	"testing"
+
+	"github.com/ViaQ/loki-operator/internal/handlers/internal/secrets"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func TestExtract(t *testing.T) {
+	type test struct {
+		name    string
+		secret  *corev1.Secret
+		wantErr bool
+	}
+	table := []test{
+		{
+			name:    "missing endpoint",
+			secret:  &corev1.Secret{},
+			wantErr: true,
+		},
+		{
+			name: "missing bucketnames",
+			secret: &corev1.Secret{
+				Data: map[string][]byte{
+					"endpoint": []byte("here"),
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "missing access_key_id",
+			secret: &corev1.Secret{
+				Data: map[string][]byte{
+					"endpoint":    []byte("here"),
+					"bucketnames": []byte("this,that"),
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "missing access_key_secret",
+			secret: &corev1.Secret{
+				Data: map[string][]byte{
+					"endpoint":      []byte("here"),
+					"bucketnames":   []byte("this,that"),
+					"access_key_id": []byte("id"),
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "all set",
+			secret: &corev1.Secret{
+				Data: map[string][]byte{
+					"endpoint":          []byte("here"),
+					"bucketnames":       []byte("this,that"),
+					"access_key_id":     []byte("id"),
+					"access_key_secret": []byte("secret"),
+				},
+			},
+		},
+	}
+	for _, tst := range table {
+		tst := tst
+		t.Run(tst.name, func(t *testing.T) {
+			t.Parallel()
+
+			_, err := secrets.Extract(tst.secret)
+			if err == nil && tst.wantErr {
+				t.Errorf("missing error: %s", err)
```

Comment:
`err` is always nil here so there's no need to print it. Also, you're not testing when `tst.wantErr` is false, but `err` is not nil